### DISCUSSION TITLE
Getting started for a CoAP contribution

### DISCUSSION THREAD

**Discussion by jvermillard** [openthread/openthread#9916] [General]
Context: I'm working on the server side of a device using open thread CoAP to communicate. The device CoAP ack timeout is not optimal for the server I work with. The device developer pointed me to this project, showing me the lack of CoAP send parameters for the CoAP secure API, so I created https://github.com/openthread/openthread/issues/9908

The code looks straightforward, I need to "port" what was done for ot_coap to ot_coapSecure

It's something I can contribute, but I don't have a specific hardware target, and I'm not interested in testing Thread, just the CoAP part.

What is the best way for me to develop and test the CoAP sub-system without having to setup a whole Thread hardware setup?

**Comment by jwhui** [comment]
OpenThread supports simulation, which is used to support both development and testing (e.g. in continuous integration tests vai GitHub Actions).
- https://openthread.io/codelabs/openthread-simulation
- https://openthread.io/codelabs/openthread-simulation-posix
